gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py

```diff
@@ -25,57 +22,63 @@ def __init__(self, *lazy_tensors, preconditioner_override=None):
             raise RuntimeError("One of the LazyTensors input to AddedDiagLazyTensor must be a DiagLazyTensor!")
 
     def inv_quad_logdet(self, inv_quad_rhs=None, logdet=False, reduce_inv_quad=True):
-        # we want to call the standard InvQuadLogDet to easily get the probe vectors and do the
-        # solve but we only want to cache the probe vectors for the backwards
-        inv_quad_term, _ = super().inv_quad_logdet(
-            inv_quad_rhs=inv_quad_rhs, logdet=False, reduce_inv_quad=reduce_inv_quad
-        )
-
-        if logdet is not False:
-            logdet_term = self._logdet()
-        else:
-            logdet_term = None
-
-        return inv_quad_term, logdet_term
+        if isinstance(self.diag_tensor, ConstantDiagLazyTensor):
+            # we want to call the standard InvQuadLogDet to easily get the probe vectors and do the
+            # solve but we only want to cache the probe vectors for the backwards
+            inv_quad_term, _ = super().inv_quad_logdet(
+                inv_quad_rhs=inv_quad_rhs, logdet=False, reduce_inv_quad=reduce_inv_quad
+            )
+            logdet_term = self._logdet() if logdet else None
+            return inv_quad_term, logdet_term
+        return super().inv_quad_logdet(inv_quad_rhs=None, logdet=False, reduce_inv_quad=True)
```

Comment:
```suggestion
        return super().inv_quad_logdet(inv_quad_rhs=inv_quad_rhs, logdet=logdet, reduce_inv_quad=reduce_inv_quad)
```

This is causing most of the test failures. Fixing this locally resolves all but one of the failures for me, where it seems like `gpytorch...LazyTensor._quad_form_derivative(...)` is missing the gradient wrt the diagonal.

Reply:
Any idea why? I don't really see the difference compared to the previous `ConstantDiagLazyTensor`.

Reply:
Not sure yet, but it doesn't appear if it's a `DiagLazyTensor`. Will next take a look through the changes to `ConstantDiagLazyTensor`.